src/dopri.ts

```diff
@@ -35,31 +40,34 @@ export class Dopri implements Integrator {
     private _lastError: number = 0;
 
     constructor(rhs: RhsFn, n: number,
-                control: Partial<DopriControlParam> = {}) {
+                control: Partial<DopriControlParam> = {},
+                output: OutputFn = null) {
         this._stepper = new dopri5.Dopri5(rhs, n);
         this._control = dopriControl(control);
+        this._output = output;
     }
 
     public initialise(t: number, y: number[]): Dopri {
         const n = this._stepper.n;
         if (y.length !== n) {
             throw Error(`Invalid size 'y' - expected a length ${n} array`);
         }
-        this._stepper.reset(y);
+        this._stepper.reset(t, y);
         this._reset();
-        this._h = this._stepper.initialStepSize(t, this._control.atol,
-                                                this._control.rtol);
+        this._h = initialStepSize(this._stepper, t, y,
+                                  this._control.atol, this._control.rtol);
         this._t = t;
+        this._history = [];
         return this;
     }
 
     public run(tEnd: number) {
-        const ret = new interpolator.Interpolator(this._stepper);
         while (this._t < tEnd) {
             this._step();
-            ret.add(this._stepper.history);
+            this._history.push(this._stepper.history.clone());
```

Comment:
You might already know this but you could replace this custom clone function with:
```
this._history.push({...this._stepper.history, data: [...this._stepper.history.data]})
```
Actually in this case maybe the clone function is clearer, and `array.splice` might be more performant than `[...data]` so probably leave as is. Just thought I'd point it out in case you haven't come across the spread operator!

Reply:
This is new to me, thanks - looks very useful